@jckarter CI fails on Linux:

~~~
******************** TEST 'Swift(linux-x86_64) :: Constraints/associated_types.swift' FAILED ********************
16:03:29 Script:
16:03:29 --
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/buildbot_linux/swift-linux-x86_64/bin/swift -frontend -target x86_64-unknown-linux-gnu  -module-cache-path '/tmp/swift-testsuite-clang-module-cacheGjbc9a' -typecheck -verify -disable-objc-attr-requires-foundation-module /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift
16:03:29 --
16:03:29 Exit Code: 1
16:03:29 
16:03:29 Command Output (stderr):
16:03:29 --
16:03:29 
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:44:20: error: unexpected note produced: protocol requires nested type 'X'; do you want to add it?
16:03:29     associatedtype X
16:03:29                    ^
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:45:20: error: unexpected note produced: protocol requires nested type 'Y'; do you want to add it?
16:03:29     associatedtype Y
16:03:29                    ^
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:55:8: error: unexpected error produced: type 'UsesSameTypedDefault' does not conform to protocol 'SameTypedDefault'
16:03:29 struct UsesSameTypedDefault: SameTypedDefault {
16:03:29        ^
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:65:20: error: unexpected note produced: protocol requires nested type 'X'; do you want to add it?
16:03:29     associatedtype X: XReqt // expected-note{{}}
16:03:29                    ^
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:66:20: error: unexpected note produced: protocol requires nested type 'Y'; do you want to add it?
16:03:29     associatedtype Y: YReqt // expected-note{{}}
16:03:29                    ^
16:03:29 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-swift-3.1-branch/swift/test/Constraints/associated_types.swift:79:8: error: unexpected error produced: type 'UsesSameTypedDefaultWithReqts' does not conform to protocol 'SameTypedDefaultWithReqts'
16:03:29 struct UsesSameTypedDefaultWithReqts: SameTypedDefaultWithReqts {
16:03:29        
~~~

--- a/include/swift/AST/TypeMatcher.h
+++ b/include/swift/AST/TypeMatcher.h
@@ -213,20 +213,7 @@ class TypeMatcher {
     TRIVIAL_CASE(DynamicSelfType)
     TRIVIAL_CASE(ArchetypeType)
     TRIVIAL_CASE(GenericTypeParamType)
-
-    bool visitDependentMemberType(CanDependentMemberType firstDepMember,
-                                  Type secondType) {
-      if (auto secondDepMember = secondType->getAs<DependentMemberType>()) {
-        if (firstDepMember->getAssocType() != secondDepMember->getAssocType() ||
-            firstDepMember->getName() != secondDepMember->getName())
-          return mismatch(firstDepMember.getPointer(), secondDepMember);
-
-        return this->visit(firstDepMember.getBase(),
-                           secondDepMember->getBase());
-      }
-
-      return mismatch(firstDepMember.getPointer(), secondType);
-    }
+    TRIVIAL_CASE(DependentMemberType)
 
     /// FIXME: Split this out into cases?
     bool visitAnyFunctionType(CanAnyFunctionType firstFunc, Type secondType) {

